### PR TITLE
chore: remove delete request body

### PR DESCRIPTION
**Pull request by KatieArmstr**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Remove the body from delete requests.

### CODE REVIEW

**Review by snaftaly — APPROVED** [review]

File: src/transmitter/index.ts [hunk 1]
```diff
@@ -308,7 +306,12 @@ export async function retryRequest(
     let statusCode: number | undefined = undefined;
 
     try {
-      response = await needle(verb, url, payload, options);
+      if (payload) {
+        response = await needle(verb, url, payload, options);
+      } else {
+        response = await needle(verb, url, options);
```
Comment (ivanstanev) [inline src/transmitter/index.ts:312]: That's a very scary function signature! How does it know we're passing "options" and not "a payload with no options"? 😵

Comment (kat1906) [reply src/transmitter/index.ts:312]: yes you are right, I will change

Comment (kat1906) [reply src/transmitter/index.ts:312]: not sure about get requests though 🤔

**Review by ivanstanev — APPROVED** [review]

**Pull request merged**